pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -252,24 +255,40 @@ func (r *Reconciler) checkDeployments(ctx context.Context, manifest *mf.Manifest
 }
 
 // ensureFinalizer attaches a "delete manifest" finalizer to the instance
-func (r *Reconciler) ensureFinalizer(ctx context.Context, manifest *mf.Manifest, instance *servingv1alpha1.KnativeServing) error {
-	for _, finalizer := range instance.GetFinalizers() {
-		if finalizer == finalizerName {
-			return nil
-		}
+func (r *Reconciler) ensureFinalizerRemoval(_ context.Context, _ *mf.Manifest, instance *servingv1alpha1.KnativeServing) error {
```

Comment:
If I understand correctly, the comment of this function should be changed into 
```
ensureFinalizerRemovel ensures that Finalizers in the operator Serving CR is consistent with the ones generated by genreconciler.
```
The `oldFinalizerName = "delete-knative-serving-manifest"` is removed, if it exists in old operator serving CR.

In the `delete` function, we do not even check the finilizers any more.

Reply:
Good catch, changed the comment.